### ISSUE TITLE
limit_val_batches and limit_train_batches do not work with argparse and an IterableDataSet

### ISSUE THREAD

**Issue by oprb** [PyTorchLightning/pytorch-lightning#12678]
## 🐛 Bug

I tried to limit train and val batches by setting the corresponding parameters via an argument passed to the script and read by `python
Trainer.from_argparse_args(args, ...)`. To be precise, I specified among other parameters`--limit_train_batches=50 --limit_val_batches=20`.I would expect that `50` and `20` are interpreted as integers (as described in the documentation) but  received the following error: "pytorch_lightning.utilities.exceptions.MisconfigurationException: When using an IterableDataset for `limit_train_batches`, `Trainer(limit_train_batches)` must be `0.0`, `1.0` or an int. An int k specifies `num_training_batches` to use."
When I specify both parameters directly in the code (`Trainer.from_argparse_args(args, ..., limit_val_batches=20, limit_train_batches=50)`), everything works as expected.
I expect that `limit_test_batches` and `limit_predict_batches` also don't work as expected when those are set as script arguments.

### To Reproduce

It should be sufficient to feed a Trainer instance by passing the arguments to the script.

- PyTorch Lightning Version: 1.5.10
- PyTorch Version: 1.9.0+cu111
- Python version: 3.8
- OS: Ubuntu Linux
- How you installed PyTorch: pip

cc @borda

**Comment by awaelchli** [comment]
Hello
Thanks for reporting this. 
My hunch is, what is happening here is simply because the argparser converts your integer inputs to floats. The implementation probably needs to be come a bit smarter here.

**Comment by oprb** [comment]
I checked the types inside the args Namespace object, `limit_val_batches` and `limit_train_batches` are both ints, `limit_test_batches` and `limit_predict_batches` are both floats (`1.0` per default, since not specified).
Unfortunately, I changed my code since I opened the issue and when I looked for the types, I had no problems specifying `limit_train_batches` and `limit_val_batches` as script arguments. Maybe something else went wrong and there is no real issue with the implementation. If I observe the problem again, I write again.
Thanks for your fast reply!

**Comment by oprb** [comment]
I checked the types inside the args Namespace object, `limit_val_batches` and `limit_train_batches` are both ints, `limit_test_batches` and `limit_predict_batches` are both floats (`1.0` per default, since not specified).
Unfortunately, I changed my code since I opened the issue and when I looked for the types, I had no problems specifying `limit_train_batches` and `limit_val_batches` as script arguments. Maybe something else went wrong and there is no real issue with the implementation. If I observe the problem again, I write again. I only noticed that the epoch progress bar shows `50.0`/`20.0` instead of `50`/`20`.

Thanks for your fast reply!
